src/GafferScene/MeshNormals.cpp

```diff
@@ -0,0 +1,174 @@
+	addChild( new FloatPlug( "thresholdAngle", Plug::In, 40, /* min */ 0.0f, /* max */ 180.0f ) );
+	addChild( new StringPlug( "position", Plug::In, "P" ) );
+	addChild( new StringPlug( "normal", Plug::In, "N" ) );
+}
+
+MeshNormals::~MeshNormals()
+{
+}
+
+Gaffer::IntPlug *MeshNormals::interpolationPlug()
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::IntPlug *MeshNormals::interpolationPlug() const
+{
+	return getChild<IntPlug>( g_firstPlugIndex );
+}
+
+Gaffer::IntPlug *MeshNormals::weightingPlug()
+{
+	return getChild<IntPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::IntPlug *MeshNormals::weightingPlug() const
+{
+	return getChild<IntPlug>( g_firstPlugIndex + 1 );
+}
+
+Gaffer::FloatPlug *MeshNormals::thresholdAnglePlug()
+{
+	return getChild<FloatPlug>( g_firstPlugIndex + 2 );
+}
+
+const Gaffer::FloatPlug *MeshNormals::thresholdAnglePlug() const
+{
+	return getChild<FloatPlug>( g_firstPlugIndex + 2 );
+}
+
+Gaffer::StringPlug *MeshNormals::positionPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
+const Gaffer::StringPlug *MeshNormals::positionPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 3 );
+}
+
+Gaffer::StringPlug *MeshNormals::normalPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 4 );
+}
+
+const Gaffer::StringPlug *MeshNormals::normalPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 4 );
+}
+
+bool MeshNormals::affectsProcessedObject( const Gaffer::Plug *input ) const
+{
+	return
+		ObjectProcessor::affectsProcessedObject( input ) ||
+		input == interpolationPlug() ||
+		input == weightingPlug() ||
+		input == thresholdAnglePlug() ||
+		input == positionPlug() ||
+		input == normalPlug()
+	;
+}
+
+void MeshNormals::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ObjectProcessor::hashProcessedObject( path, context, h );
+	interpolationPlug()->hash( h );
+	weightingPlug()->hash( h );
+	thresholdAnglePlug()->hash( h );
+	positionPlug()->hash( h );
+	normalPlug()->hash( h );
+}
+
+IECore::ConstObjectPtr MeshNormals::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, const IECore::Object *inputObject ) const
+{
+	const MeshPrimitive *mesh = runTimeCast<const MeshPrimitive>( inputObject );
+	if( !mesh )
+	{
+		return inputObject;
+	}
+
+	std::string normal = normalPlug()->getValue();
+	PrimitiveVariable::Interpolation interpolation = (PrimitiveVariable::Interpolation) interpolationPlug()->getValue();
+	MeshAlgo::NormalWeighting weighting = (MeshAlgo::NormalWeighting) weightingPlug()->getValue();
+	float thresholdAngle = thresholdAnglePlug()->getValue();
+	std::string position = positionPlug()->getValue();
+
+	MeshPrimitivePtr meshWithNormals = runTimeCast<MeshPrimitive>( mesh->copy() );
+
+	if( interpolation == PrimitiveVariable::Uniform )
+	{
+		meshWithNormals->variables[ normal ] = MeshAlgo::calculateUniformNormals(
```

Comment:
You've overridden `computeCachePolicy()` for `out.object`, buts that's not the plug that we're doing the processing on. The processing is done on the `__processedObject` plug, so that we don't incur the expense of TaskCollaboration for pass-throughs on the `out.object` plug. As per the comment above, we need to override `processedObjectComputeCachePolicy()` instead.

If you could also rebase this on main at the same time, it should pick up the Cortex update with the necessary MeshAlgo additions.